### PR TITLE
include: sensor: Add position change channel and make sensor API more flexible

### PR DESCRIPTION
**Pull request by pdunaj**
This would be used for motion sensors.

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

**Comment by pdunaj** [comment]
hi @carlescufi , @anangl , we need this fast :)

**Comment by pdunaj** [comment]
Hi @anangl , @MaureenHelm , I will also need to pass some attributes to sensor. The problem is I think the current API is rather poorly designed as I need to put a sensor specific value in common sensor.h file.
I would like to add an extension that would allow us to get the attribute from include file specific for the sensor. Extension as I don't have time to rework the current API (plus maybe somebody like or need it as is).

**Comment by carlescufi** [comment]
@pdunaj check out this branch: https://github.com/zephyrproject-rtos/zephyr/tree/topic-sensors for the rework that is currently ongoing with the sensor API, which has a lot of activity.

**Comment by pdunaj** [comment]
Hi @carlescufi , great but we need to add our change now. Also I think this API can be saved with very small effort - I will send a PR for that in a sec.

**Comment by carlescufi** [comment]
> Hi @carlescufi , great but we need to add our change now. Also I think this API can be saved with very small effort - I will send a PR for that in a sec.

Sure, I am not saying that we wait for the new sensor API, we can of course add the features we need to the old one, especially if they are non-intrusive like the change you propose.

**Comment by pdunaj** [comment]
I have pushed it here. I think it would do the trick if we allow higher values to be passed as trig/chan/attr so sensor can have its private things defined in the header file (e.g. `SENSOR_ATTR_MYATTR = SENSOR_ATTR_COUNT + 1`).

This change require that enums are converted to u32_t or some more generic type so we always have place for higher values. Will add it if you are happy with the proposal.

@MaureenHelm , @anangl , @carlescufi

**Review by anangl — APPROVED** [review]
LGTM

File: include/sensor.h [hunk 1]
```diff
@@ -125,8 +125,32 @@ enum sensor_channel {
...
+	 * This and higher values are sensor specific.
+	 * Refer to the sensor header file.
+	 */
+	SENSOR_CHAN_PRIV_START = SENSOR_CHAN_COMMON_COUNT,
+
+	/**
+	 * Maximum value describing a sensor channel type.
+	 */
+	SENSOR_CHAN_MAX = INT16_MAX,
+
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply include/sensor.h]: Fixed, thanks

**Comment by pdunaj** [comment]
Hi @carlescufi , can you merge?

**Comment by carlescufi** [comment]
@pdunaj I would really like for @MaureenHelm to take a look before.

**Comment by pdunaj** [comment]
Hi @carlescufi , me too but we have 2 other PRs waiting for this change + upmerge will be needed. I would prefer not waiting days for it.

Hi @MaureenHelm , can you have a look?

**Comment by pdunaj** [comment]
Hi @carlescufi , I will create a PR downstream. BTW - would it be possible to assign at least 2 code owners for any module so if one is very busy or don't care there is some fallback?

**Review by MaureenHelm — APPROVED** [review]

**Comment by pdunaj** [comment]
thanks

**Pull request merged**